### PR TITLE
[ISSUE #6767] optimize the push data logic.

### LINKED ISSUE #6767: Optimize the push data generate logic.
**Describe the bug**

move push data generate logic to each client(stub)

https://github.com/alibaba/nacos/blob/4ac1a9b0ed8b892708a67cf7418583d349b91369/naming/src/main/java/com/alibaba/nacos/naming/push/v2/task/PushExecuteTask.java#L55-L80

**Expected behavior**

move the push data generate logic to each client.

**Acutally behavior**

**How to Reproduce**

**Desktop (please complete the following information):**
 - OS: [e.g. Centos]
 - Version [nacos 2.0.3]
 - Module [naming]

**Additional context**
no

### PR DESCRIPTION
**Pull request by brotherlu-xcq**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Fixes #6767 

## Brief changelog

- move the select instance logic to push task.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutor.java [hunk 1]
```diff
@@ -30,19 +31,21 @@
     /**
      * Do push.
      *
+     * @param service    the origin service
      * @param clientId   client id
      * @param subscriber subscriber
      * @param data       push data
      */
-    void doPush(String clientId, Subscriber subscriber, PushDataWrapper data);
+    void doPush(Service service, String clientId, Subscriber subscriber, PushDataWrapper data);
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutor.java:39]: Add Service into PushDataWrapper will be better.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java [hunk 2]
```diff
@@ -34,19 +39,26 @@
     
     private final RpcPushService pushService;
     
-    public PushExecutorRpcImpl(RpcPushService pushService) {
+    private final NamingMetadataManager metadataManager;
+    
+    public PushExecutorRpcImpl(RpcPushService pushService, NamingMetadataManager metadataManager) {
         this.pushService = pushService;
+        this.metadataManager = metadataManager;
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java:46]: Also service metadata add into pushdata wrapper?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutor.java (hunk 1 above)
Comment (brotherlu-xcq) [reply naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutor.java:39]: OK, I will move it later.

File: naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java (hunk 2 above)
Comment (brotherlu-xcq) [reply naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java:46]: I think PushdataWrapper only keep the data is better. metaDataManager like a global DB, if we put in this wrapper, user maybe not know update the metadataManager will change the global data.

Comment (KomachiSion) [reply naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java:45]: metadata and service both the relative data of push data. So saved in wrapper has no problem.

If you are afraid that the metadata or service will be changed in executors, you can clone a new one to cache.

If you input `metadataManager`, there may be two problem I has known.

1. In `PushExecutorDelegate`, we have an SPI for executors, if your push data not include metadata, but data selector need metdata, means all executors SPI must depend `metadataManager`, which may cause you change the executors SPI to add an method to inject `metadataManager`. Not a good design. 
2. For one push task, all subscribers should receive same data, if you input `metadataManager`, during the push task, the metadata changed, which will cause the push data maybe not same for one push task.

Comment (brotherlu-xcq) [reply naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java:45]: Yes, agree. I will change it.

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**